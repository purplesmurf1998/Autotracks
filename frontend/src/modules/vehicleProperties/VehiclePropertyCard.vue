<template>
  <CCard>
    <CCardBody class="p-3">
      <view-property
        v-if="!editingProperty"
        :property="property"
        :setEditingProperty="setEditingProperty"
      />
      <edit-property 
        v-if="editingProperty"
        :setEditingProperty="setEditingProperty"
        :property="property"
        :index="index"
        :updateProperty="updateProperty"
      />
    </CCardBody>
  </CCard>
</template>

<script>
import ViewVehicleProperty from "./ViewVehicleProperty.vue";
import EditVehicleProperty from "./EditVehicleProperty.vue"

export default {
  name: 'VehiclePropertyCard',
  data() {
    return {};
  },
  props: ["property", "editingProperty", "setEditingProperty", 'index', 'updateProperty'],
  components: {
    "view-property": ViewVehicleProperty,
    "edit-property": EditVehicleProperty
  }
};
</script>

<style scoped>
</style>