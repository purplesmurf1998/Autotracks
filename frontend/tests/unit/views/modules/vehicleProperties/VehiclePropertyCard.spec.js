import {createLocalVue, shallowMount} from '@vue/test-utils'
import CoreuiVue from '@coreui/vue-pro'
import Vuex from 'vuex'

import VehiclePropertyCard from './../../../../../src/modules/vehicleProperties/VehiclePropertyCard.vue'

const localVue = createLocalVue()
localVue.use(CoreuiVue)
localVue.use(Vuex)

const store = new Vuex.Store({
    state: {
      sidebarShow: 'responsive',
      sidebarMinimize: false,
      asideShow: false,
      darkMode: false
    }
  })

  describe('VehiclePropertyCard.vue', () => {
    const wrapper = shallowMount(VehiclePropertyCard, { 
      store, 
      localVue
    })
    it('has a name', () => {
      expect(VehiclePropertyCard.name).toBe('VehiclePropertyCard')
    })
    it('should render correct content', () => {
        
    })
    test('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })