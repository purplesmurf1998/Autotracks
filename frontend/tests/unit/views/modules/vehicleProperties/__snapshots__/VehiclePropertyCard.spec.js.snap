// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehiclePropertyCard.vue renders correctly 1`] = `
<ccard-stub>
  <ccardbody-stub
    class="p-3"
  >
    <view-property-stub />
     
    <!---->
  </ccardbody-stub>
</ccard-stub>
`;
